src/mumble/pluginAPI/PluginComponents.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * This header file contains definitions of types and other components used in Mumble's plugin system
+ */
+
+#ifndef MUMBLE_PLUGINCOMPONENT_H_
+#define MUMBLE_PLUGINCOMPONENT_H_
+
+#include <stdint.h>
+
+#define STATUS_OK 0
+
+struct Version {
+	int32_t major;
+	int32_t minor;
+	int32_t patch;
+};
+
+
+struct MumbleUser {
+	const char *name;
+	uint32_t id;
+};
+
+struct MumbleChannel {
+	const char *name;
+	int32_t id;
+};
+
+
+typedef struct Version Version_t;
+typedef struct MumbleUser MumbleUser_t;
+typedef struct MumbleChannel MumbleChannel_t;
+
+typedef int32_t MumbleConnection_t
+
+
+struct MumbleAPI {
+	// -------- Memory (de-)allocation --------
+	
+	/// Allocates memory for a MumbleUser. Once the user struct is no longer needed, it has to be freed by calling
+	/// freeMumbleUser.
+	///
+	/// @returns A pointer to the allocated struct.
+	MumbleUser_t* (*allocateMumbleUser)();
+
+	/// Frees a MumbleUser that has been allocated via allocateMumbleUser.
+	///
+	/// @params user A pointer to the user struct that should be freed
+	void (*freeMumbleUser)(MumbleUser_t *user);
+
+
+	/// Allocates memory for a MumbleChannel. Once the channel struct is no longer needed, it has to be freed by calling
+	/// freeMumbleChannel.
+	///
+	/// @returns A pointer to the allocated struct.
+	MumbleChannel_t* (*allocateMumbleChannel)();
+
+	/// Frees a MumbleChannel that has been allocated via allocateMumbleChannel.
+	///
+	/// @param channel A pointer to the channel struct that should get freed
+	void (*freeMumbleChannel)(MumbleChannel_t *channel);
+
+	
+
+	
+	// -------- Getter functions --------
+
+	/// Fills in the connection ID of the server the user is currently active on (the user's audio output is directed at).
+	///
+	/// @param[out] connection A pointer to the memory location the ID should be written to
+	/// @returns The error code. If everything went well, STATUS_OK will be returned. Only then it is valid to access the
+	/// 	value of the provided pointer
+	uint32_t getActiveServerConnection(MumbleConnection_t *connection);
+
+	/// Fills in the information about the local user.
+	///
+	/// @param connection A reference to which server-connection should be considered.
```

Comment:
API currently says it's a copy of the struct, not a reference.
Why would we ever always want to pass the id AND the name? id should be enough.

Reply:
What name? You provide an allocated user struct whose fields will be set by this function

Reply:
Yes we pass the struct, which contains ID and name. But what for? Isn't the plain userID all that's needed here?

Reply:
The struct doesn't contain the ID and the name at this point. Those fields will be set by this function. That's its purpose.

Reply:
I'm talking about the connection, not the user.

Reply:
The connection isn't even a struct. It's just an int